In [2]:
import math as m
import numpy as np
import scipy as sc
import pandas as pd

### Задача 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [73]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

In [74]:
cov = np.mean(zp * ks) - np.mean(zp) * np.mean(ks)
cov

9157.839999999997

In [75]:
np.cov(zp, ks, ddof=0)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

In [76]:
# Коэфф. Пирсона

In [77]:
np.std(zp)

59.115480206118605

In [78]:
np.std(ks)

174.55340157098058

In [79]:
9157.84 / (59.115480206118605 * 174.55340157098058)

0.8874900920739162

In [80]:
np.corrcoef(zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [85]:
df = pd.DataFrame({'a': [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], 
                   'b': [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]})
a

a    b
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [84]:
df['a'].corr(df['b'])

0.8874900920739163

### Задача 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

[Xср - tα/2 * sigma/sqrt(n); Xср + tα/2 * sigma/sqrt(n)]

In [88]:
x = [131, 125, 115, 122, 131, 115, 107, 99, 125, 111]
Xср = np.mean(x)
Xср

118.1

In [89]:
sigma = np.std(x, ddof=1)
sigma

10.54566788359614

In [90]:
Xср - 2.262 * sigma / m.sqrt(len(x))

110.55660776308164

In [91]:
Xср + 2.262 * sigma / m.sqrt(len(x))

125.64339223691834

Ответ: [110.55660776308164; 125.64339223691834]

### Задача 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

[Xср - Zα/2 * sigma/sqrt(n); Xср + Zα/2 * sigma/sqrt(n)]

In [95]:
Xср = 174.2
Z = 1.96
n = 27
sigma = m.sqrt(25)

In [97]:
Xср - Z * sigma/m.sqrt(n)

172.31398912064722

In [98]:
Xср + Z * sigma/m.sqrt(n)

176.08601087935276

Ответ: [172.31398912064722; 176.08601087935276]